src/gromacs/nblib/topology.cpp

```diff
@@ -127,31 +148,35 @@ TopologyBuilder& TopologyBuilder::addMolecule(const Molecule molecule, const int
     return *this;
 }
 
-const std::vector<real>& Topology::getMasses() const
+const std::vector<real>& Topology::masses() const
 {
-    return masses;
+    return masses_;
```

Comment:
Here and below it is a coding error to return this if no masses have been set so we should add asserts that masses_, etc are not size 0.

Reply:
I don't think it's a coding error. It's not the responsibility of this observer to guarantee that the state of the object is correct.
We do have the TopologyBuilder and a private constructor in Topology to ensure that a Topology object can only be created in a valid state.